### PR TITLE
Support separate CSP configs for Mozorg and Pocket modes

### LINKED ISSUE #11543: Frontend-specific improvements for Pocket mode: CSP config
## Description

Following on from #11494's comments there are some frontend-specific changes that it would be good to make.

Sepecfically, we could have a separate CSP config for both mozorg vs pocket mode
 
---

#### Success Criteria
- [ ] Pocket runs its own CSP config, separate from that for Mozorg

### PR DESCRIPTION
**Pull request by stevejalim**
## One-line summary

This changeset splits our current CSP config into two, so that we can apply either Mozorg-specific or Pocket-specific rules, based on the active site mode

## Significant changes and points to review

* When splitting out the CSP rules, I've removed pocket-specific ones from the mozorg config
* I'd welcome another view on if the configs are correct
* In Pocket mode, the devtools console shows a CORS issue when using localhost - can we do something about this?
  
## Issue / Bugzilla link

Resolves #11543

## Testing

* `$ npm start` then view http://localhost:8080. The page should load with all assets (you may have to wait for webpack) Check the devtools console - there should be no issues
* * `$ npm run in-pocket-mode` then view http://localhost:8080. Check the devtools console - there should be no issues, but there is a CORS niggle, noted above.

### CODE REVIEW

**Review by robhudson — APPROVED** [review]
Looks like a good clean-up. Is the general idea that site mode based settings should be in the `__init__.py` file?

**Comment by stevejalim** [comment]
@robhudson 

> Is the general idea that site mode based settings should be in the `__init__.py` file?

I think _for now_, yes, but I'd like to eventually squash it all down to a single `settings.py`, if that doesn't seem too large to work with easily. What d'you think?

**Review by maureenlholland — APPROVED** [review]
r+ LGTM, I like the separation, sorry I can't be more help on the snowplow CORS error

File: bedrock/settings/__init__.py [hunk 1]
```diff
@@ -179,10 +104,136 @@
...
     LANGUAGE_CODE = "en"  # Pocket uses `en` not `en-US`
 
+    # CSP settings for POCKET, expanded upon later:
+    _csp_default_src = [
+        "'self'",
+        "*.getpocket.com",
+    ]
+    _csp_img_src = [
+        "data:",
+        "www.googletagmanager.com",
```
Comment (alexgibson) [inline bedrock/settings/__init__.py:114]: Does Pocket use GTM?

I see calls to `www.google-analytics.com` when viewing https://getpocket.com/add, but not `www.googletagmanager.com"`.

File: bedrock/settings/__init__.py [hunk 2]
```diff
@@ -179,10 +104,136 @@
...
+    ]
+    _csp_script_src = [
+        # TODO fix use of OptanonWrapper() so that we don't need this
+        "'unsafe-inline'",
+        # TODO onetrust cookie consent breaks
+        # blocked without unsafe-eval. Find a way to remove that.
+        "'unsafe-eval'",
+        "www.googletagmanager.com",
+        "www.google-analytics.com",
+        "tagmanager.google.com",
```
Comment (alexgibson) [inline bedrock/settings/__init__.py:125]: Same question as above, can we remove `www.googletagmanager.com` and `tagmanager.google.com`?

File: bedrock/settings/__init__.py [hunk 3]
```diff
@@ -179,10 +104,136 @@
...
+    ]
+    _csp_style_src = [
+        "'unsafe-inline'",
+    ]
+    _csp_child_src = []
+    _csp_connect_src = [
+        "www.googletagmanager.com",
+        "www.google-analytics.com",
+        "o1069899.sentry.io",
+        "o1069899.ingest.sentry.io",
```
Comment (alexgibson) [inline bedrock/settings/__init__.py:137]: Are we using Sentry for Pocket? Happy to leave this in if that is the plan, but if not we can probably remove these?

File: bedrock/settings/__init__.py [hunk 4]
```diff
@@ -179,10 +104,136 @@
...
+        "cdn.cookielaw.org",
+        "privacyportal.onetrust.com",
+        "getpocket.com",  # Pocket Snowplow
+        "geolocation.onetrust.com",
+    ]
+    _csp_connect_extra_for_dev = [
+        "com-getpocket-prod1.mini.snplow.net",
```
[the remark this answers is not in the document]
Comment (alexgibson) [reply bedrock/settings/__init__.py:144]: Fwiw, I see this on the main branch as well. My guess is their CORS policy disallows `localhost` for these requests?

**Review by alexgibson — APPROVED** [review]
A few open questions about what's needed and where, but I'm not 100% sure on the answers (glad we're picking this apart now and not in another year's time 😅)

Both modes seem to work well for me, r+ (pending whatever might need updating)

File: bedrock/settings/__init__.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (maureenlholland) [reply bedrock/settings/__init__.py:144]: Good point - yes, this is an existing issue so not a blocker for the current PR. Maybe we just don't make the snowplow tracker call at all on dev mode?

File: bedrock/settings/__init__.py (hunk 1 above)
Comment (stevejalim) [reply bedrock/settings/__init__.py:114]: It's possible we don't need it - we could zap it now, or wait and zap it once we're sure it's not used in production. It's a domain that we trust for Mozorg, at least.

Comment (alexgibson) [reply bedrock/settings/__init__.py:114]: Up to you here if / when we want to remove stuff, but from my perspective I think it's good to keep the lists lean and separate so we can really restrict each site to only what's needed.

File: bedrock/settings/__init__.py [hunk 5]
```diff
@@ -179,10 +104,136 @@
...
+    ]
+    _csp_img_src = [
+        "data:",
+        "mozilla.org",
+        "www.googletagmanager.com",
+        "www.google-analytics.com",
+        "adservice.google.com",
+        "adservice.google.de",
+        "adservice.google.dk",
+        "creativecommons.org",
```
[the remark this answers is not in the document]
Comment (stevejalim) [reply bedrock/settings/__init__.py:172]: It's from Mozorg's history it seems - https://github.com/mozilla/bedrock/commit/3bedbf1fd59184de15a36d607503ed44293cedfb

Whether we need it, though...

File: bedrock/settings/__init__.py (hunk 1 above)
Comment (alexgibson) [reply bedrock/settings/__init__.py:114]: If you want we can always file a "clean up the dual-site CSP mish-mash" issue for after merge? (using a more sensible name of course 😅)

Comment (stevejalim) [reply bedrock/settings/__init__.py:114]: Zapping now - we can add back in if pre-launch testing shows we should

File: bedrock/settings/__init__.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (alexgibson) [reply bedrock/settings/__init__.py:172]: Ah, found it: https://github.com/mozilla/bedrock/blob/67fd925b4d621dde0e48e27738c4b9c397ac5169/bedrock/foundation/templates/foundation/licensing/website-content.html#L22

Looks like we do need it (maybe for this solitary image, lol)

**Pull request merged**